src/clientSideScene/setup.tsx

```diff
@@ -0,0 +1,1280 @@
+              className="text-black w-16"
+              onChange={(e) => {
+                setupSingleton.setCam({
+                  ...camSettings,
+                  position: [
+                    camSettings.position[0],
+                    parseFloat(e.target.value),
+                    camSettings.position[2],
+                  ],
+                })
+              }}
+            />
+          </li>
+          <li>
+            <span className="pl-2 pr-1">z:</span>
+            <input
+              type="number"
+              step={5}
+              data-testid="cam-z-position"
+              value={camSettings.position[2]}
+              className="text-black w-16"
+              onChange={(e) => {
+                setupSingleton.setCam({
+                  ...camSettings,
+                  position: [
+                    camSettings.position[0],
+                    camSettings.position[1],
+                    parseFloat(e.target.value),
+                  ],
+                })
+              }}
+            />
+          </li>
+        </ul>
+      </div>
+    </div>
+  )
+}
+
+function convertThreeCamValuesToEngineCam({
+  position,
+  quaternion,
+  zoom,
+  isPerspective,
+}: ThreeCamValues): {
+  center: Vector3
+  up: Vector3
+  vantage: Vector3
+} {
+  // Something to consider is that the orbit controls have a target,
+  // we're kind of deriving the target/lookAtVector here when it might not be needed
+  // leaving for now since it's working but maybe revisit later
+  const euler = new Euler().setFromQuaternion(quaternion, 'XYZ')
+
+  const lookAtVector = new Vector3(0, 0, -1)
+    .applyEuler(euler)
+    .normalize()
+    .add(position)
+
+  const upVector = new Vector3(0, 1, 0).applyEuler(euler).normalize()
+  if (isPerspective) {
+    return {
+      center: lookAtVector,
+      up: upVector,
+      vantage: position,
+    }
+  }
+  const zoomFactor = -ZOOM_MAGIC_NUMBER / zoom
+  const direction = lookAtVector.clone().sub(position).normalize()
+  const newVantage = position.clone().add(direction.multiplyScalar(zoomFactor))
+  return {
+    center: lookAtVector,
+    up: upVector,
+    vantage: newVantage,
+  }
+}
+
+function calculateNearFarFromFOV(fov: number) {
+  const nearFarRatio = (fov - 3) / (45 - 3)
+  const z_near = 0.1 + nearFarRatio * (5 - 0.1)
+  const z_far = 1000 + nearFarRatio * (100000 - 1000)
+  return { z_near: 0.1, z_far }
+}
+
+function updateEngineFov(args: {
+  position: Vector3
+  quaternion: Quaternion
+  zoom: number
+  fov: number
+}) {
+  engineCommandManager.sendSceneCommand(
+    {
+      type: 'modeling_cmd_req',
+      cmd_id: uuidv4(),
+      cmd: {
+        type: 'default_camera_perspective_settings',
+        ...convertThreeCamValuesToEngineCam({
+          ...args,
+          isPerspective: true,
+        }),
+        fov_y: args.fov,
+        ...calculateNearFarFromFOV(args.fov),
+      },
+    } as any /* TODO - this command isn't in the spec yet, remove any when it is */
```

Comment:
I'll remove the `any` and comment once `https://github.com/KittyCAD/engine/pull/1760` is merged and kittycad.ts is updated.